client/db/src/lib.rs

```diff
@@ -55,8 +55,11 @@ use std::{
 use crate::{
 	stats::StateUsageStats,
 	storage_cache::{new_shared_cache, CachingState, SharedCache, SyncingCachingState},
-	utils::{meta_keys, read_db, read_meta, DatabaseType, Meta},
+	utils::{meta_keys, read_db, Meta},
 };
+
+pub use crate::utils::{read_meta, open_database, DatabaseType};
```

Comment:
Would be better to not expose these internals.
If I look at code from client/service/src/builder.rs, I see:
```
		let db_config = sc_client_db::DatabaseSettings {
			state_cache_size: config.state_cache_size,
			state_cache_child_ratio: config.state_cache_child_ratio.map(|v| (v, 100)),
			state_pruning: config.state_pruning.clone(),
			source: config.database.clone(),
			keep_blocks: config.keep_blocks.clone(),
		};

		let backend = new_db_backend(db_config)?;
```
then call ```backend.blockchain().info()``` (requires some trait imports).

Reply:
That's way better. Let me change it.

Reply:
I did not know about this, otherwise I would have use it as well :laughing: